sklearn/discriminant_analysis.py

```diff
@@ -649,12 +651,10 @@ def fit(self, X, y, store_covariances=None, tol=None):
             Target values (integers)
         """
         if store_covariances:
-            warnings.warn("The parameter 'store_covariances' is deprecated as "
-                          "of version 0.17 and will be removed in 0.19. The "
-                          "parameter is no longer necessary because the value "
-                          "is set via the estimator initialisation or "
-                          "set_params method.", DeprecationWarning)
-            self.store_covariances = store_covariances
+            warnings.warn("'store_covariances' was renamed to 'store_covariance'"
```

Comment:
You shouldn't change the warning message here. Carefully reading the earlier message shows that this statement won't be there from 0.19 onwards and hence, this is just added to ensure compatibility until 0.19. So, your warning message is not valid. The best you could do is add a note alongside the one already present as:
`Also, 'store_covariances' is renamed to 'store_covariance' in 0.19.`